File: app/src/main/java/com/danis/nadi/data/db/entity/ChatMessageEntity.java
package com.danis.nadi.data.db.entity;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import com.danis.nadi.model.ChatMessage;

import androidx.room.Ignore;

@Entity(tableName = "chat_messages")
public class ChatMessageEntity {
    @PrimaryKey
    @NonNull
    public String messageId;
    
    @NonNull
    public String roomId;
    
    @NonNull
    public String senderId;
    
    @NonNull
    public String senderName;
    
    @NonNull
    public String text;
    
    public long createdAt;
    
    @NonNull
    public String status;
    
    @Nullable
    public String attachmentTransferId;
    
    @Nullable
    public String attachmentFileName;
    
    @Nullable
    public String attachmentMimeType;
    
    public long attachmentSizeBytes = -1L;
    
    @NonNull
    public String attachmentStatus = "";

    public ChatMessageEntity() {}

    @Ignore
    public ChatMessageEntity(@NonNull String messageId, @NonNull String roomId, @NonNull String senderId,
                             @NonNull String senderName, @NonNull String text, long createdAt, @NonNull String status,
                             @Nullable String attachmentTransferId, @Nullable String attachmentFileName,
                             @Nullable String attachmentMimeType, long attachmentSizeBytes, @NonNull String attachmentStatus) {
        this.messageId = messageId;
        this.roomId = roomId;
        this.senderId = senderId;
        this.senderName = senderName;
        this.text = text;
        this.createdAt = createdAt;
        this.status = status;
        this.attachmentTransferId = attachmentTransferId;
        this.attachmentFileName = attachmentFileName;
        this.attachmentMimeType = attachmentMimeType;
        this.attachmentSizeBytes = attachmentSizeBytes;
        this.attachmentStatus = attachmentStatus;
    }

    public ChatMessage toDomain() {
        return new ChatMessage(
            messageId,
            senderId,
            senderName,
            text,
            createdAt,
            status,
            attachmentTransferId,
            attachmentFileName,
            attachmentMimeType,
            attachmentSizeBytes,
            attachmentStatus
        );
    }

    public static ChatMessageEntity fromDomain(ChatMessage domain, String roomId) {
        return new ChatMessageEntity(
            domain.getMessageId(),
            roomId,
            domain.getSenderId(),
            domain.getSenderName(),
            domain.getText(),
            domain.getCreatedAt(),
            domain.getStatus(),
            domain.getAttachmentTransferId(),
            domain.getAttachmentFileName(),
            domain.getAttachmentMimeType(),
            domain.getAttachmentSizeBytes(),
            domain.getAttachmentStatus()
        );
    }
}

File: app/src/main/java/com/danis/nadi/data/db/entity/ChatMessageEntity.kt
package com.danis.nadi.data.db.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.danis.nadi.model.ChatMessage

@Entity(tableName = "chat_messages")
data class ChatMessageEntity(
    @PrimaryKey val messageId: String,
    val roomId: String,
    val senderId: String,
    val senderName: String,
    val text: String,
    val createdAt: Long,
    val status: String,
    val attachmentTransferId: String? = null,
    val attachmentFileName: String? = null,
    val attachmentMimeType: String? = null,
    val attachmentSizeBytes: Long = -1L,
    val attachmentStatus: String = ""
) {
    fun toDomain(): ChatMessage {
        return ChatMessage(
            messageId = messageId,
            senderId = senderId,
            senderName = senderName,
            text = text,
            createdAt = createdAt,
            status = status,
            attachmentTransferId = attachmentTransferId,
            attachmentFileName = attachmentFileName,
            attachmentMimeType = attachmentMimeType,
            attachmentSizeBytes = attachmentSizeBytes,
            attachmentStatus = attachmentStatus
        )
    }

    companion object {
        fun fromDomain(domain: ChatMessage, roomId: String): ChatMessageEntity {
            return ChatMessageEntity(
                messageId = domain.messageId,
                roomId = roomId,
                senderId = domain.senderId,
                senderName = domain.senderName,
                text = domain.text,
                createdAt = domain.createdAt,
                status = domain.status,
                attachmentTransferId = domain.attachmentTransferId,
                attachmentFileName = domain.attachmentFileName,
                attachmentMimeType = domain.attachmentMimeType,
                attachmentSizeBytes = domain.attachmentSizeBytes,
                attachmentStatus = domain.attachmentStatus
            )
        }
    }
}
